cogito/put.go

```diff
@@ -136,21 +138,67 @@ func (putter *ProdPutter) LoadConfiguration(in io.Reader, args []string) error {
 }
 
 func (putter *ProdPutter) ProcessInputDir() error {
-	inputDirs, err := collectInputDirs(putter.InputDir)
+	// putter.InputDir, corresponding to key "put:inputs:", should contain 1 or 2 dirs.
+	// If it contains one, we support autodiscovery by not requiring to name it, we know
+	// that it should be the git repo.
+	// If on the other hand it contains two, one should be the git repo (still nameless)
+	// and the other should be the directory containing the chat_message_file, which is
+	// named by the first element of the path in "chat_message_file".
+	// This allows (although clumsily) to distinguish which is which.
+	// This complexity has historical reasons to preserve backwards compatibility
+	// (the nameless git repo).
+	//
+	// Somehow independent is the reason why we enforce the count of directories to be
+	// max 2: this is to avoid the default Concourse behavior of streaming _all_ the
+	// volumes "just in case".
+
+	params := putter.Request.Params
+	source := putter.Request.Source
+	var msgDir string
+
+	if params.ChatMessageFile != "" {
+		msgDir, _ = path.Split(params.ChatMessageFile)
+		msgDir = strings.TrimSuffix(msgDir, "/")
+		if msgDir == "" {
+			return fmt.Errorf("chat_message_file: wrong format: have: %s, want: path of the form: <dir>/<file>",
+				params.ChatMessageFile)
+		}
+	}
+
+	collected, err := collectInputDirs(putter.InputDir)
```

Comment:
Could this block be placed before L159? it'd permit to check earlier the conditions and consequently less `if/else` if I'm not wrong

Reply:
@Esysc what is the "block"? Only lines 168--171 or more?

Reply:
Never mind, I see you addressed in https://github.com/Pix4D/cogito/pull/100/commits/855c81916ffa98c8c856b055347714e1c0a99fc5